backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/description/EdgeDescription.java

```diff
@@ -92,16 +91,16 @@ public SynchronizationPolicy getSynchronizationPolicy() {
         return this.semanticElementsProvider;
     }
 
-    public Function<VariableManager, Optional<Label>> getBeginLabelProvider() {
-        return this.beginLabelProvider;
+    public Optional<LabelDescription> getOptionalBeginLabelDescription() {
+        return this.optionalBeginLabelDescription;
     }
 
-    public Function<VariableManager, Optional<Label>> getCenterLabelProvider() {
-        return this.centerLabelProvider;
+    public Optional<LabelDescription> getOptionalCenterLabelDescription() {
+        return this.optionalCenterLabelDescription;
     }
 
-    public Function<VariableManager, Optional<Label>> getEndLabelProvider() {
-        return this.endLabelProvider;
+    public Optional<LabelDescription> getOptionalEndLabelDescription() {
+        return this.optionalEndLabelDescription;
```

Comment:
No need to make optional part of the API like that, other properties can be null. Use something like this:

```
public LabelDescription getBeginLabelDescription() {
  return this.beginLabelDescription;
}
```